### PR TITLE
Fix Teleport Method

### PR DESCRIPTION
**Pull request by louis1706**
Using NW method is not bad :) it's does what we have needed here also fix people teleporting in wall or in object or in the void

### CODE REVIEW

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -2778,7 +2778,7 @@ public void Reconnect(ushort newPort = 0, float delay = 5, bool reconnect = true
         /// Teleports the player to the given <see cref="Vector3"/> coordinates.
         /// </summary>
         /// <param name="position">The <see cref="Vector3"/> coordinates to move the player to.</param>
-        public void Teleport(Vector3 position) => Position = position;
+        public void Teleport(Vector3 position) => ReferenceHub.playerMovementSync.OverridePosition(position);
```
Comment (IRacle1) [inline Exiled.API/Features/Player.cs:2781]: Better to just edit `Position` setter

Comment (louis1706) [reply Exiled.API/Features/Player.cs]: no :)

File: Exiled.API/Features/Player.cs [hunk 2]
```diff
@@ -2778,7 +2778,12 @@ public void Reconnect(ushort newPort = 0, float delay = 5, bool reconnect = true
         /// Teleports the player to the given <see cref="Vector3"/> coordinates.
         /// </summary>
         /// <param name="position">The <see cref="Vector3"/> coordinates to move the player to.</param>
-        public void Teleport(Vector3 position) => Position = position;
+        public void Teleport(Vector3 position)
+        {
+            if (!PlayerMovementSync.FindSafePosition(position, out Vector3 safepos, true, true))
+                safepos = position;
+            ReferenceHub.playerMovementSync.OverridePosition(safepos);
```
Comment (iopietro) [inline Exiled.API/Features/Player.cs:2785]:
```suggestion
            if (!PlayerMovementSync.FindSafePosition(position, out Vector3 safepos, true, true))
                safepos = position;

            ReferenceHub.playerMovementSync.OverridePosition(safepos);
```

**Review by iopietro — CHANGES_REQUESTED** [review]
Should we add the `isSafe` optional parameter to safely or not safely spawn the player at the chosen coordinates?

**Comment by Thundermaker300** [comment]
Target dev branch.

**Comment by louis1706** [comment]
I close this PR Because FindSafePosition no longer exist and nothing similar than this exist

**Pull request closed without merging**